pkg/agentctl/sync.go

```diff
@@ -0,0 +1,122 @@
+package agentctl
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/client"
+	"github.com/grafana/agent/pkg/prometheus/instance"
+	"gopkg.in/yaml.v2"
+)
+
+// ConfigSync loads YAML files from a directory and syncs them to the
+// provided PrometheusClient API. All YAML files will be synced and
+// must be valid.
+//
+// The base name of the YAML file (i.e., without the file extension)
+// is used as the config name.
+//
+// ConfigSync will completely overwrite the set of active configs
+// present in the provided PrometheusClient - configs present in the
+// API but not in the directory will be deleted.
+func ConfigSync(logger log.Logger, cli client.PrometheusClient, dir string) error {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+
+	ctx := context.Background()
+	existing, err := cli.ListConfigs(ctx)
+	if err != nil {
+		return fmt.Errorf("could not list configs: %w", err)
+	}
+
+	cfgs, err := ConfigsFromDirectory(dir)
+	if err != nil {
+		return err
+	}
+
+	uploaded := map[string]struct{}{}
```

Comment:
```suggestion
	uploaded := make(map[string]struct{}, 0, len(cfgs)) 
```

Reply:
FYI when you `make` a map with a size like this, it doesn't support a capacity argument like slices do and will error from the compiler. It should be `make(map[string]struct{}, len(cfgs))`. I missed this at first but I'll fix it in a commit

Reply:
Yep! You're right 🤦 - sorry for missing that!